src/with_threads_alloc.rs

```diff
@@ -0,0 +1,171 @@
+use crate::ringbuffer_trait::RingBuffer;
+
+extern crate alloc;
+// We need vecs so depend on alloc
+use crate::{ReadableRingbuffer, WritableRingbuffer};
+use alloc::vec::Vec;
+
+/// The AllocRingBuffer is a RingBuffer which is based on a Vec. This means it allocates at runtime
+/// on the heap, and therefore needs the [`alloc`] crate. This struct and therefore the dependency on
+/// alloc can be disabled by disabling the `alloc` (default) feature.
+///
+/// # Example
+/// ```rust
+/// // TODO: Example
+/// ```
+#[derive(PartialEq, Eq, Debug)]
+pub struct ThreadAllocRingBuffer<T> {
```

Comment:
I think we want to add this in the future so for now I suggest we branch from this branch, remove the threaded code an make a new PR?